apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -871,77 +794,55 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     :one
   end
 
-  def fetch_datastream_maximum_storage_retention(client) do
-    maximum_storage_retention_statement = """
-    SELECT blobAsInt(value)
-    FROM kv_store
-    WHERE group='realm_config' AND key='datastream_maximum_storage_retention'
-    """
+  def fetch_datastream_maximum_storage_retention(realm) do
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
 
     query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(maximum_storage_retention_statement)
-      |> DatabaseQuery.consistency(:quorum)
+      KvStore
+      |> where(group: "realm_config", key: "datastream_maximum_storage_retention")
+      |> select([v], fragment("blobAsInt(?)", v.value))
+      |> put_query_prefix(keyspace_name)
 
```

Comment:
super nit: Instead of directly handling special case `KvStore` fetching here you can rely on [`astarte_data_access#17 astarte/kvstore.ex`](https://github.com/noaccOS/astarte_data_access/blob/ae7573b0183a66090c8a091044a0a9b2e3170773/lib/astarte_data_access/astarte/kv_store.ex#L87-L102) `fetch_value` function and handle all these kinds of castings

Reply:
After IRL discussion, we've decided to wait before going with this solution.